x-pack/platform/plugins/shared/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -307,7 +307,7 @@ export class ObservabilityAIAssistantClient {
                 disableFunctions,
                 tracer: completeTracer,
                 connectorId,
-                useSimulatedFunctionCalling: simulateFunctionCalling === true,
+                useSimulatedFunctionCalling: functionCallingMode === 'simulated',
```

Comment:
Instead of having two options `useSimulatedFunctionCalling` and `functionCallingMode` perhaps just pass `functionCallingMode` through:

```suggestion
                functionCallingMode,
```

Reply:
++, I would keep the ternary option down in the chain if possible

Reply:
Thank you for the review @sorenlouv @pgayvallet 
I will pass `functionCallingMode` as suggested and remove `useSimulatedFunctionCalling`